indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/TaskLimiterUtils.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.overlord.setup;
+
+import org.apache.druid.indexing.common.task.Task;
+
+import java.util.Map;
+
+public class TaskLimiterUtils
+{
+
+  /**
+   * Determines whether the given task can be executed based on task limits and available capacity.
+   *
+   * @param task The task to evaluate.
+   * @param taskLimits The limits and ratios defining task execution constraints.
+   * @param currentSlotsUsed The current capacity used by tasks of the same type.
+   * @param totalCapacity The total available capacity across all workers.
+   * @return {@code true} if the task meets the defined limits and capacity constraints; {@code false} otherwise.
+   */
+  public static boolean canRunTask(Task task, TaskLimits taskLimits, Integer currentSlotsUsed, Integer totalCapacity)
```

Comment:
I think this need not be a static utility method.
This could very well live as a non-static method in `TaskLimits` itself.

```
public boolean canRunTask(Task task, Integer currentSlotsUsed, Integer totalCapacity) {
}
```